src/BindPattern.jl

```diff
@@ -79,10 +79,10 @@ function bind_pattern!(
         pattern = BoundEqualValueTestPattern(
             location, source, input, source, ImmutableDict{Symbol, Symbol}())
 
-    elseif (source isa Expr && source.head == :$)
+    elseif source isa Expr && source.head == :$
         # an interpolation
         interpolation = source.args[1]
-        (interpolation0, assigned0) = subst_patvars(interpolation, assigned)
+        interpolation0, assigned0 = subst_patvars(interpolation, assigned)
```

Comment:
I didn't know you could do that.  I guess `,` has higher precedence than `=`, which is unexpected.